MantidQt/MantidWidgets/src/InstrumentView/InstrumentActor.cpp

```diff
@@ -757,9 +761,9 @@ void InstrumentActor::setupPickColors() {
 void InstrumentActor::cacheDetPos() const {
   if (m_detPos.size() != m_detIDs.size()) {
     m_detPos.clear();
-    for (size_t i = 0; i < m_detIDs.size(); i++) {
-      IDetector_const_sptr det = this->getDetector(i);
-      m_detPos.push_back(det->getPos());
+    for (size_t pickID = 0; pickID < m_detIDs.size(); pickID++) {
```

Comment:
Good point about time steps. I guess there are two options: (1) keep `pickID` as a linear index that can then map to detectors with time index (a `std::pair` of detector index and time index, see class `SpectrumDefinition`), or (2) eventually add a time index to `pickID`.

I do not like option (2) since it is exposing something about the instrument that does not need to be known at this level. Maybe, for hiding the fact that there are time indices, would it make sense to use `SpectrumInfo`? Such a `SpectrumInfo` could be set up with a 1:1 map from spectrum to `pair(detector index, time index)` (i.e., this could be different from a mapping in a workspace). We would then have a plain linear index, without the need to mess around with time indices.